### PR TITLE
Default backup/restore for all nifi deploys

### PR DESCRIPTION
**Pull request by gstuder-ona**
Had some of this work from earlier tests on unicef-rwanda.

Fixes: https://github.com/onaio/canopy/issues/1015, https://github.com/onaio/canopy/issues/827

Adds backup jobs to all nifi deploys unless explicitly disabled.  Backups will be local unless an S3 link with credentials is specified.  See the README changes.

Two outstanding issues:
1) Ideally we should use nifi-tools file-manager.sh to do the non-flowfile backup - this can be added as an enhancement if we like here (tools are downloaded, but aren't used)
2) We need to take down the nifi service to get a non-corrupt flowfile snapshot - maybe this shouldn't be the default to prevent a sharp edge?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,10 +9,7 @@ Role has been tested with NiFi versions 1.1.x, 1.2.0, 1.3.0 and 1.4.0.
 Requirements
 ------------
 
-Prior to executing this role, the NiFi distribution must be accssible on the target system at
-```{{ nifi_base_dir }}/nifi-{{ nifi_version }}/```
-  - if RPM, the RPM must be installed
-  - if tar.gz, it must be unarchived
+  - `molecule[docker]` v3 or compatible
```
Comment (gstuder-ona) [inline README.md:12]: Ran into an issue during testing - our molecule version here is different from playbooks' version - we should probably update playbooks to molecule v3.

File: molecule/backup-restore/Dockerfile.minio.j2 [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Molecule managed
```
Comment (gstuder-ona) [inline molecule/backup-restore/Dockerfile.minio.j2:1]: for testing S3 backup

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+[tool.poetry]
```
Comment (gstuder-ona) [inline pyproject.toml:1]: I don't really care what tool we use but we probably want to define our dependencies.  This was fastest for me.

**Review by pld — APPROVED** [review]

**Comment by pld** [comment]
others w/comments on this please add, we can continue in another PR

**Pull request merged**